sdk/azidentity/example_cache_test.go

```diff
@@ -0,0 +1,32 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity_test
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+
+	// importing the cache module registers the cache implementation for the current platform
```

Comment:
Might be good to augment this comment stating that Linux variants aren't supported at present.

Reply:
Sorry, I should have written in the description that I won't ship this without Linux support. My goal with this PR is to review everything else, to avoid a last minute review bomb or at least make it a smaller bomb.